internal/querywatcher/query_watcher.go

```diff
@@ -188,17 +200,68 @@ func (w *QueryManager) updateQueryCache(queryFingerprint string, event dstore.Wa
 func (w *QueryManager) notifyClients(query *sql.DSQLQuery, clients *sync.Map, queryResult *[]sql.QueryResultRow) {
 	encodedResult := clientio.Encode(clientio.CreatePushResponse(query, queryResult), false)
 
-	clients.Range(func(clientKey, _ interface{}) bool {
-		clientFD := clientKey.(int)
-		// We use a retry mechanism here as the client's socket may be temporarily unavailable for writes due to the
-		// high number of writes that are possible in qwatch. Without this mechanism, the client may be removed from the
-		// watchlist prematurely.
-		// TODO:
-		//  1. Replace with thread pool to prevent launching an unbounded number of goroutines.
-		//  2. Each client's writes should be sent in a serialized manner, maybe a per-client queue should be maintained
-		//   here. A single queue-per-client is also helpful when the client file descriptor is closed and the queue can
-		//   just be destroyed.
-		go w.sendWithRetry(query, clientFD, encodedResult)
+	clients.Range(func(clientKey, clientVal interface{}) bool {
+		// Identify the type of client and respond accordingly
+		// Client response channel can be currently of type response writer and
+		switch clientResponseChannel := clientKey.(type) {
+		case http.ResponseWriter:
+			rp := clientio.NewRESPParser(bytes.NewBuffer(encodedResult))
+			val, err := rp.DecodeOne()
+			if err != nil {
+				log.Errorf("Error decoding response: %v", err)
+				return true
+			}
+
+			var responseJSON []byte
+			// Convert the decoded response to the HTTPQwatchResponse struct
+			// Else just encode and send
+			switch v := val.(type) {
+			case []interface{}:
+				if len(v) >= 3 {
+					qwatchResp := HTTPQwatchResponse{
+						Cmd:   v[0].(string),
+						Query: v[1].(string),
+						Data:  v[2].([]interface{}),
+					}
+					responseJSON, err = json.Marshal(qwatchResp)
+				}
+			default:
+				responseJSON, err = json.Marshal(val)
```

Comment:
Will this ever get executed? Just wondering

Reply:
Even if it doesn't, we should probably return an error response - saying "unimplemented protocol" or something similar to that so that we don't silently ignore such issues.